clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -473,12 +473,30 @@
                                                                        new Field("max_bytes",
                                                                                  INT32,
                                                                                  "Maximum bytes to fetch."));
+    // FETCH_REQUEST_PARTITION_V1 added log_start_offset field - the smallest available offset of partition data that can be consumed.
+    public static final Schema FETCH_REQUEST_PARTITION_V1 = new Schema(new Field("partition",
+                                                                                 INT32,
+                                                                                 "Topic partition id."),
+                                                                       new Field("fetch_offset",
+                                                                                 INT64,
+                                                                                 "Message offset."),
+                                                                       new Field("log_start_offset",
+                                                                                 INT64,
+                                                                                 "Smallest available offset."),
```

Comment:
Nit: In consumer we use the term "earliest" instead of "smallest", maybe we can make the term consistent.